packages/backend/src/controllers/organizationController.ts

```diff
@@ -0,0 +1,298 @@
+    /**
+     * Get the current user's organization
+     * @param req express request
+     */
+    @Middlewares([allowApiKeyAuthentication, isAuthenticated])
+    @Get()
+    @OperationId('getOrganization')
+    async getOrganization(
+        @Request() req: express.Request,
+    ): Promise<ApiOrganization> {
+        this.setStatus(200);
+        return {
+            status: 'ok',
+            results: await organizationService.get(req.user!),
+        };
+    }
+
+    /**
+     * Update the current user's organization
+     * @param req express request
+     * @param body the new organization settings
+     */
+    @Middlewares([isAuthenticated, unauthorisedInDemo])
+    @Patch()
+    @OperationId('updateOrganization')
+    async updateOrganization(
+        @Request() req: express.Request,
+        @Body() body: UpdateOrganization,
+    ): Promise<ApiSuccessEmpty> {
+        await organizationService.updateOrg(req.user!, body);
+        this.setStatus(200);
+        return {
+            status: 'ok',
+            results: undefined,
+        };
+    }
+
+    /**
+     * Delete an organization and all users inside that organization
+     * @param req express request
+     * @param organizationUuid the uuid of the organization to delete
+     */
+    @Middlewares([isAuthenticated, unauthorisedInDemo])
+    @Delete('{organizationUuid}')
+    @OperationId('deleteOrganization')
+    async deleteOrganization(
+        @Request() req: express.Request,
+        @Path() organizationUuid: string,
+    ): Promise<ApiSuccessEmpty> {
+        await organizationService.delete(organizationUuid, req.user!);
+        await promisify(req.session.destroy)();
+        this.setStatus(200);
+        return {
+            status: 'ok',
+            results: undefined,
+        };
+    }
+
+    /**
+     * Get all projects for the current user's organization
+     */
+    @Middlewares([allowApiKeyAuthentication, isAuthenticated])
+    @Get('/projects')
+    @OperationId('getOrganizationProjects')
+    async getOrganizationProjects(
+        @Request() req: express.Request,
+    ): Promise<ApiOrganizationProjects> {
+        this.setStatus(200);
+        return {
+            status: 'ok',
+            results: await organizationService.getProjects(req.user!),
+        };
+    }
+
+    /**
+     * Create a new project in the current user's organization. Compile the project for the user.
+     * @param req express request
+     * @param body the project to create
+     * @param method the method used to create the project (CLI, CLI_CI, WEB_APP, UNKNOWN)
+     */
+    @Middlewares([
+        allowApiKeyAuthentication,
+        isAuthenticated,
+        unauthorisedInDemo,
+    ])
+    @Post('/projects/precompiled')
+    @OperationId('createPrecompiledProject')
+    async createPrecompiledProject(
+        @Request() req: express.Request,
+        @Body() body: CreateProject,
+        @Header(LightdashRequestMethodHeader) method: RequestMethod,
```

Comment:
Perhaps we should type this as a potential `undefined` and default to `RequestMethod.UNKNOWN` in `projectService.create()`